src/OutputAreaViewModel.ts

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IChangedArgs, Property
+} from 'phosphor-properties';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  ObservableList
+} from 'phosphor-observablelist';
+
+/**
+* A Mime bundle of data.
+*/
+export interface MimeBundle {
+  [key: string]: string;
+  'application/json'?: any;
+}
+
+/**
+* The base interface for an output view model.
+*/
+export
+class OutputBaseViewModel {
+
+  /**
+  * A signal emitted when state of the output changes.
+  */
+  stateChanged: ISignal<OutputBaseViewModel, IChangedArgs<any>>;
+
+  /**
+  * The output type.
+  */
+  outputType: string; // "execute_result" | "display_data" | "stream" | "error"
```

Comment:
Enum all the way!

On Sat, Dec 5, 2015 at 12:59 PM, Jason Grout <notifications@github.com>
wrote:

> In src/OutputAreaViewModel.ts
> <https://github.com/jupyter/jupyter-js-output-area/pull/24#discussion_r46762935>
> :
>
> > +
> > +/**
> > +* The base interface for an output view model.
> > +*/
> > +export
> > +class OutputBaseViewModel {
> > +
> > +  /**
> > +  * A signal emitted when state of the output changes.
> > +  */
> > +  stateChanged: ISignal<OutputBaseViewModel, IChangedArgs<any>>;
> > +
> > +  /**
> > +  * The output type.
> > +  */
> > +  outputType: string; // "execute_result" | "display_data" | "stream" | "error"
>
> I guess an enum provides typing information now, and takes less space. We
> should probably be consistent between using enums vs strings for this sort
> of typing. @ellisonbg <https://github.com/ellisonbg>, do you have a
> preference?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-output-area/pull/24/files#r46762935>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com